cmake/ActsCompilerOptions.cmake

```diff
@@ -2,10 +2,23 @@
 if (NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type configuration" FORCE)
   message(STATUS "Setting default build type: ${CMAKE_BUILD_TYPE}")
-endif() 
+endif()
+
+set(extra_flags "-Wall;-Wextra;-Wpedantic;-Wshadow;-Wno-unused-local-typedefs")
```

Comment:
You're doing quite some back-and-forth conversion here. :confused: Why not do the whole thing with "string operations"?

I.e.

```cmake
set(extra_flags "-Wall -Wextra -Wpedantic -Wshadow -Wno-unused-local-typedefs")
if(...)
   set(extra_flags "${extra_flags} -Wfloat-conversion")
endif()
```

Then you can add the flags coming "from the outside" with the same formalism as well.

Reply:
I guess I was trying to avoid the extra space, but might not be important.